src/Approximations/overapproximate.jl

```diff
@@ -300,23 +299,23 @@ Overapproximating a set with template directions.
 
 ### Output
 
-An `HPolytope` overapproximating the set `X` with the directions from `dir`.
+A polyhedron overapproximating the set `X` with the directions from `dir`.
+If the directions are known to be bounded, the result is an `HPolytope`,
+otherwise the result is an `HPolyhedron`.
 """
-function overapproximate(X::LazySet{N},
-                         dir::AbstractDirections{N}
-                        )::HPolytope{N} where {N}
+function overapproximate(X::LazySet{N}, dir::AbstractDirections{N}) where {N}
     halfspaces = Vector{LinearConstraint{N}}()
     sizehint!(halfspaces, length(dir))
-    H = HPolytope(halfspaces)
+    T = isbounding(dir) ? HPolytope : HPolyhedron
```

Comment:
I thought if this check should rather be done through dispatch on the type of`dir`, but i don't know if this choice impacts performance in a considerable way.

Reply:
This should be in the order of milliseconds. Maybe the compiler can even add this information statically.

Reply:
FYI, this line has been removed with the new changes.